clang/lib/Sema/HLSLExternalSemaSource.cpp

```diff
@@ -540,10 +555,10 @@ struct BuiltinTypeMethodBuilder {
 
     // create method decl
     auto *TSInfo = AST.getTrivialTypeSourceInfo(MethodTy, SourceLocation());
-    Method =
-        CXXMethodDecl::Create(AST, DeclBuilder.Record, SourceLocation(),
-                              NameInfo, MethodTy, TSInfo, SC_None, false, false,
-                              ConstexprSpecKind::Unspecified, SourceLocation());
+    Method = CXXMethodDecl::Create(
+        AST, DeclBuilder.finalizeForwardDeclaration(), SourceLocation(),
```

Comment:
This change is not necessary - methods are added when the type is getting completed on its first use, which is long after the forward declaration was created.